Automattic/jetpack
# Jetpack Social: Add caching to shares info

#### Changes proposed in this Pull Request:

* Add caching to the shares info in the Publicize package, and return a cached response if the API fails.

#### Other information:


#### Jetpack product discussion


1202583289493814-as-1203047881068112/f

#### Does this pull request change what data or activity we track or use?


No.

#### Testing instructions:






You can use the following command to get the response from the `get_publicize_shares_info` method: `echo "global \$publicize; return \$publicize->get_publicize_shares_info(Jetpack_Options::get_option( 'id' ));" | docker exec -i jetpack_dev_wordpress_1 wp --allow-root shell`

* Add a non-existent sandbox to your local config: `define( 'JETPACK__SANDBOX_DOMAIN', 'example.com' );`
* Run the command above and confirm it gives an error.
* Remove the sandbox constant and run the command again.
* Confirm it doesn't give an error anymore.
* Now, add the sandbox constant again and rerun the command.
* Confirm it still doesn't give an error, which means it's using the cached version.
